src/lib/tools.ts

```diff
@@ -181,22 +251,21 @@ export async function unwrapKeyframe(
  * @returns {Promise<any>}
  */
 export async function runInTunnel(
-	tnlConfig: any,
-	prom: any,
+	tnlConfig: tunnel.Config,
+	prom: Promise<any>,
 	timeout: number,
 ): Promise<any> {
-	return await tunnelAsync(tnlConfig).then(tnl => {
-		const wait = setTimeout(function() {
+	const tnl = await tunnelAsync(tnlConfig);
+	const wait = setTimeout(function() {
+		tnl.close();
+		throw new TimeoutError('Timeout exceeded');
+	}, timeout);
+	return prom.then((ret: any) => {
+		clearTimeout(wait);
+		if (tnl) {
```

Comment:
Don't think there's much point checking for `tnl` here; if it can legally return a falsey value, then the `tnl.close()` in the `wait` function isn't checking correctly. If it can't, then the tunnel has failed anyway.